### PR TITLE
CLI initial version

### PR DESCRIPTION
**Pull request by carletex**
**NOTE:** Further changes to this branch should be done by a PR

> from the work we did on https://github.com/carletex/create-dapp-example (real-se2-templates / feat/extension-templateConifg-dgrcode branches)

This PR changes the workflow for users to use scaffold-eth-2. Instead of cloning, users will use this CLI (npx create-whatever-name@latest). It'll let users pick different options (hardhat/foundry + extensions)

image

Initially, we'll start with hardhat/foundry (so we can merge and iterate), but @dgrcode built an extension system. Read more [here](https://github.com/scaffold-eth/scaffold-eth-2/blob/99a8a8db198812b0c2ce867e5e129d85359357d8/RFC-extensions.md)

Some ToDos (maybe we should create some issues?):
- [x] Update README (add new way of installing it, dev notes for the cli, etc)
- [x] GH Action to auto-publish to npm (extra: auto-increment version on release/merge)
- [ ] Check node version (=> 18) and yarn installed before running. We could also have a "dependencies" file (or in config.json) for each extension, so we check them before installing (e.g. foundry extension needs to have foundryup in the system)
- [x] ~~Better ASCII art (or just chalk with color and fancy chars) on the intro message. (render-intro-message.ts)~~ Fixed https://github.com/scaffold-eth/scaffold-eth-2/pull/413/commits/42e26bb18c8247b614ac6cd5a4b22433cfad6828
- [x] ~~In the outro message hint about yarn chain / yarn deploy / yarn start (render-outro-message.ts)~~. Fixed https://github.com/scaffold-eth/scaffold-eth-2/pull/413/commits/da4395063eb23785695a5b778a674c0063961165
- [x] ~~Fix foundry template: Let's remove the need of having to create a .env file manually (the cli can do it for you... for now, until we find a system for extensions to add their own .env)~~ Fixed https://github.com/scaffold-eth/scaffold-eth-2/pull/413/commits/8e05bbbe92568c4e6b41dd17fe29a027053eb357
- [ ] Create the graph extension
- [ ] Submodules: Foundry uses submodules, and we are setting that up directly on the CLI. It'd be ideal to move this to the template (e.g. the template declares some submodules, and the CLI dynamically set them up for every extesion)
- [x] Contributions: Find the best way to test PRs, develop new features with this system without having to copy/paste from a "installed instance". Something like hot-reloading
- [ ] Contributions: Move to a release branch model (TLDR: instead of merging directly to main, use relase branches: eg. release-2.1.0 to batch stuff and the merge to master. Only hotfix goes directly to main)
- [ ] Add prettier / husky / linting (and maybe remove some of them from the base template)

We don't need to solve all of them before merging.

PS: There are a lot of "files changed", but most of them are just moving things to the template folders (which git is detecting :))

### CODE REVIEW

**Comment by carletex** [comment]
Went for `create-eth` which was suggested by @austintgriffith and it was free :D

`npx create-eth@latest`

Could you guys try? @sverps @rin-st @Pabl0cks @damianmarti  (not tagging @technophile-04 because he is next to me IRL rn :D)

**Comment by damianmarti** [comment]
@carletex this is awesome!! :-)

I tested it trying to think that I'm a new user of SE-2 and I think that these improvements can be good things to do:

1. Add a progress bar when installing the dependencies with yarn (it takes some time and I think that the loading is not enough, at least for me I like to see what is happening or at least a progress bar moving). I don't know how easy or hard is to add it, I like to use the command **pv** on Ubuntu for this kind of thing...
2. It's on the ToDo list, but I think that after finishing we can add a short version of the README file, and add a reference to where the user can find more information (full README file or whatever). 
3. Add some information about installing extensions and the available extensions.

I used the installed se-2 project and everything seems fine.

We are close to having the cli ready!! :partying_face:

**Comment by rin-st** [comment]
Hey, great job!

For now I'm getting an error when trying to preinstall packages for both hh and foundry (Install packages? y). Part of the log:
```
...
➤ YN0007: │ bufferutil@npm:4.0.7 must be built because it never has been before or the last one failed
➤ YN0007: │ utf-8-validate@npm:5.0.10 must be built because it never has been before or the last one failed
➤ YN0009: │ se-2@workspace:. couldn't be built successfully (exit code 1, logs can be found here: /private/var/folders/rs/t2dhts8n6hv59mn9srqyh74r0000gn/T/xfs-b42708b1/build.log)
➤ YN0009: │ deasync@npm:0.1.28 couldn't be built successfully (exit code 1, logs can be found here: /private/var/folders/rs/t2dhts8n6hv59mn9srqyh74r0000gn/T/xfs-dd82bdf5/build.log)
➤ YN0007: │ ssh2@npm:1.14.0 must be built because it never has been before or the last one failed
➤ YN0000: └ Completed in 18s 70ms
➤ YN0000: Failed with errors in 35s 91ms

    at makeError (/Users/liana/Documents/se-2/node_modules/pkg-install/node_modules/execa/index.js:174:9)
    at /Users/liana/Documents/se-2/node_modules/pkg-install/node_modules/execa/index.js:278:16
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  code: 1,
  stdout: '➤ YN0000: ┌ Resolution step\n' +
    '➤ YN0032: │ keccak@npm:3.0.3: Implicit dependencies on node-gyp are discouraged\n' +
    '➤ YN0061: │ debug@npm:3.2.6 is deprecated: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)\n' +
...
```

`build.log` files:

```
# This file contains the result of Yarn building a package (se-2@workspace:.)
# Script name: postinstall

husky - .git can't be found (see https://typicode.github.io/husky/#/?id=custom-directory)
```
and
```
# This file contains the result of Yarn building a package (deasync@npm:0.1.28)
# Script name: install

[31m[1mUsage Error[22m[39m: Couldn't find a script name "node-gyp" in the top-level (used by deasync@npm:0.1.28). This typically happens because some package depends on "node-gyp" to build itself, but didn't list it in their dependencies. To fix that, please run "yarn add node-gyp" into your top-level workspace. You also can open an issue on the repository of the specified package to suggest them to use an optional peer dependency.

[1m$ [22myarn run [--inspect] [--inspect-brk] [-T,--top-level] [-B,--binaries-only] <scriptName> ...
Build failed
```
so something wrong with `husky`, `.git` or `node-gyp`. Tried to install node-gyp and change node version to 18 but it didn't help. Will try to debug and understand how it works later today, but not sure it will be fast.

Other than that noticed that foundry project readme should be changed, but it's already mentioned. And name of the local network for foundry shouldn't be `hardhat` :)

**Comment by technophile-04** [comment]
@rin-st did you get this when doing it using `npx create-eth@latest` or you were trying it out locally using `yarn cli` ? 

Because we were getting this error when you are trying it out locally using `yarn cli` and you select the project name in the current directory (where all the cli stuff is), like you need to create it outside this project directory : 

eg : 
```
> yarn cli 

 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+
 | Create Scaffold-ETH 2 app |
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+

?  Your project name: ../my-dapp-example
```

We need to document this problem in README while working locall....But could you please confirm if you got this error while doing `npx create-eth@latest` or doing `yarn cli` locally ?

**Comment by rin-st** [comment]
@technophile-04 I have an error in both `npx create-eth@latest` and `yarn cli`. But using `../my-dapp-example` works for both cases 👍

**Comment by rin-st** [comment]
btw I was trying to use `npx create-eth@latest` from `se-2` folder. If I use it from outside git repo it works as expected

File: src/main.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      task: () => createProjectDirectory(options.project),
+    },
+    {
+      title: `🚀 Creating a new Scaffold-ETH 2 app in ${chalk.green.bold(
+        options.project
+      )}`,
+      task: () =>
+        copyTemplateFiles(options, templateDirectory, targetDirectory),
+    },
+    {
```
Comment (rin-st) [inline src/main.ts:38]: changing order to `initializing git` -> `installing deps` fixes my errors even for `yarn cli`

**Comment by Pabl0cks** [comment]
Awesome job! Was exciting to see @carletex and @technophile-04 working on this IRL 🚀

On Windows (both my desk and laptop) it's working like a charm when I choose hardhat, but when I choose foundry, I got problems to `yarn chain`. 

```
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+
 | Create Scaffold-ETH 2 app |
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+

? Your project name: test3
? What solidity framework do you want to use? foundry
? Install packages? Yes


  √ 📁 Create project directory C:\Users\Pablo\test3
  √ 🚀 Creating a new Scaffold-ETH 2 app in test3
  √ 📦 Installing dependencies with yarn, this could take a while
  √ 📡 Initializing git repository



  Congratulations! Your project has been scaffolded! 🎉

  Next steps:

  cd test3

        Start the local development node
        yarn chain

        In a new terminal window, deploy your contracts
        yarn deploy

        In a new terminal window, start the frontend
        yarn start

  Thanks for using Scaffold-ETH 2 🙏, Happy Building!


$ cd test3

 ~/test3 (main)

$ yarn chain
"anvil" no se reconoce como un comando interno o externo,
programa o archivo por lotes ejecutable.
command not found: anvil

```

**Comment by carletex** [comment]
Thanks for the feedback and the help @rin-st.

> btw I was trying to use npx create-eth@latest from se-2 folder. If I use it from outside git repo it works as expected

Yep, this was happening when running it inside an existing repo command (because of the postinstall yarn hook). Your fix works perfectly, but it leaves the `yarn.lock` file uncommitted (changed after running `yarn install`**), so we decided to take a different approach (git init before yarn install, but add & commit after yarn install) and it seems to work. @technophile-04 will push in a bit.

** This is expected since we have an empty `yarn.lock` file, to avoid conflicts when we have a bunch of extensions. So it's generated on the first install. In package.json files, we changed ^ for ~, so with packages bugfixes there is less risk of something breaking (we tried to do exact versions in package.json, but for some reason the peer-dependencies weren't installing)

> On Windows (both my desk and laptop) it's working like a charm when I choose hardhat, but when I choose foundry, I got problems to yarn chain.

Thanks @Pabl0cks ! Yes, this happens because people need to install foundryup (that contains anvil and some other tools) in their **system**. What we are going to do is check if they have it installed and if not show a message hinting them to install.

We'll be pushing (& publishing) in a bit.

Thanks all!

File: RFC-extensions.md [hunk 2]
```diff
@@ -150,16 +171,18 @@ create-dapp-example/
...
 │  │  ├─ foo/
 │  │  │  ├─ some-folder/
-│  │  │  │  ├─ template-at-root.md.template.js <-- won't work!
-│  │  │  │  ├─ template-at-folder.md.template.js
-│  │  │  ├─ template-at-root.md.template.js
-│  │  │  ├─ template-at-folder.md.template.js <-- won't work!
+│  │  │  │  ├─ template-at-root.md.args.js <-- won't work!
+│  │  │  │  ├─ template-at-folder.md.args.js
+│  │  │  ├─ template-at-root.md.args.js
+│  │  │  ├─ template-at-folder.md.args.js <-- won't work!
```
Comment (dgrcode) [inline RFC-extensions.md:177]: Good catch! 👍

**Comment by Pabl0cks** [comment]
> Thanks @Pabl0cks ! Yes, this happens because people need to install foundryup (that contains anvil and some other tools) in > their system. What we are going to do is check if they have it installed and if not show a message hinting them to install.

Thx @carletex I had no idea about that. Now with foundryup is working great!  
Getting the informative message before installing it, so I know I have to install foundryup => Installed it and tried again, and  CLI is working perfect with foundry also, got my working SE-2 🙌

I'll send a nitpick edge case:
- When selecting foundry, and selecting "_No_" at the "_Install packages?_" option, it shows a finish icon at the Git step, but I don't get any feedback or next steps after instalation. With hardhat and "No packages" I get the congrats message with the steps to start SE-2.

That's all I get with foundry:

>$ npx create-eth@latest
>
> +-+-+-+-+-+-+-+-+-+-+-+-+-+-+
> | Create Scaffold-ETH 2 app |
> +-+-+-+-+-+-+-+-+-+-+-+-+-+-+
>
>? Your project name: test9
>? What solidity framework do you want to use? foundry
>? Install packages? No
>
>
>  √ 📁 Create project directory C:\Users\Pablo\test9
>  √ 🚀 Creating a new Scaffold-ETH 2 app in test9
>  ↓ 📦 Installing dependencies with yarn, this could take a while [skipped]
>  → Manually skipped
>  √ 📡 Initializing Git repository

**Comment by carletex** [comment]
> When selecting foundry, and selecting "No" at the "Install packages?" option, it shows a finish icon at the Git step, but I don't get any feedback or next steps after instalation. With hardhat and "No packages" I get the congrats message with the steps to start SE-2.

Yes @Pabl0cks! I think Shiv solved it here: https://github.com/scaffold-eth/scaffold-eth-2/pull/413/commits/9bb9422b072fa52fa233214acd5c11516647618e

I haven't published it yet to NPM but it should work if run the CLI locally (instead of create-eth)

**Review by dgrcode — COMMENTED** [review]
I'm having a look at the PR. Some thoughts/questions:
 - Shouldn't we commit the `dist/` folder? I'm curious how npx is working without the dist folder. Do they run the build task on their servers? 🤷
 - What do you think about committing the `.vscode/` folder to keep the debugger configuration? It definitely helps me, and I guess it could help anybody trying to debug the cli.
 - I've talked with @carletex about the two files that need to be renamed: `.env-default` and `.gitignore-keep`. I'll convert them to templates so we can skip the renames and avoid exception for certain files.

TODO list for myself:
 - [ ] Use templates for renamed files
 - [ ] Remove unused types from `src/types.ts`
 - [x] Look into why `findFilesRecursiveSync` wasn't working for `.gitignore-keep` and `.env-default`. Even though we won't need to rename files, it might have a bug that's not showing up and could break something in the future.

**Comment by carletex** [comment]
@dgrcode 

> Shouldn't we commit the dist/ folder? I'm curious how npx is working without the dist folder. Do they run the build task on their servers? 

Yep, it builds there

> What do you think about committing the .vscode/ folder to keep the debugger configuration? It definitely helps me, and I guess it could help anybody trying to debug the cli.

We decided against it in SE-2. Not everyone uses VSCode (eg. I use Intellij and Shiv uses Vim). The same way that I don't want to commit the `.idea` folder from my Intelij

The ToDos look great and pretty straightforward. Also remember to take a look at the "how to contribute" stuff (hot reloading, symlinks, different repo) which is the most important piece.

**Comment by dgrcode** [comment]
> Also remember to take a look at the "how to contribute" stuff (hot reloading, symlinks, different repo) which is the most important piece

Yes, will do! but it could probably go in a different PR, no?

File: templates/base/.gitignore-keep [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+# dependencies, yarn, etc
+# yarn / eslint
+.yarn/*
+!.yarn/patches
+!.yarn/plugins
+!.yarn/releases
+!.yarn/sdks
+!.yarn/versions
+.eslintcache
+.vscode/**
```
[the remark this answers is not in the document]
Comment (dgrcode) [reply templates/base/.gitignore-keep:12]: Decided against it

**Comment by carletex** [comment]
@dgrcode 

> Yes, will do! but it could probably go in a different PR, no?

Yep!

As we also talked offline, I'm going to protect this branch in the repo settings, so people will need to make a PR before merging here.

Thanks Dani!

**Comment by dgrcode** [comment]
opened https://github.com/scaffold-eth/scaffold-eth-2/pull/417 for the bug above

**Comment by technophile-04** [comment]
Bug : It seems that `yarn next:types` after fresh install of any options. : 

Screenshot 2023-07-14 at 1 07 19 AM


To reproduce : 
1. `npx create-eth@latest` 
2. Choose `none` or any tempate
3. Yes for yarn install (you can `cd` and then do `yarn install`)
4.  Run : `yarn next:check-types`

**Comment by dgrcode** [comment]
I just opened the PR https://github.com/scaffold-eth/scaffold-eth-2/pull/423 to remove an item from my TODO list.

**Comment by schmidsi** [comment]
Wow, nice. Just ran `npx create-eth@latest` on my Mac Book Air M2, MacOS version 13.4.1 (22F82), Node version v18.15.0 and Yarn version 3.2.3 and it was smooth. I choose Foundry to check this out and it worked well. I really liked that it checked if Foundry was installed and told me how to install it: `(NOTE: Foundryup is not installed in your system)`. That is great developer experience 🤩.

Two small remarks:
- I personally don't like submodules. They can get easily messy quickly. Especially junior devs that are not familiar with it can quickly run into weird problems. But I assume that comes upstream from Foundry. Isn't there a way to run/install forge and anvill without submodules?
- `packages/nextjs/generated/deployedContracts.ts` is marked as changed. [It seems that there is no community consensus over adding generated code to source control or not.](https://stackoverflow.com/questions/893913/should-i-store-generated-code-in-source-control) I personally try to avoid adding generated code to source control. In this case, that could be solved as part of the build step, no?

**Comment by carletex** [comment]
Thanks for testing it and the feedback @schmidsi !

> I personally don't like submodules.

Same! We had some issues making them work with the cli too. But I think there is no way around it with foundry, since it's the way it manages its deps.

> packages/nextjs/generated/deployedContracts.ts is marked as changed. 

This was changed after you run `yarn deploy` right? That file contains "the artifacts" (address + abi) of the contracts deployed with SE-2, so you need them in order to interact with those contracts in the frontend. Doing it in the build step will be tricky since the source of truth lives on the hardhat package (when you deploy the front-end you only deploy the `nextjs` package). See this discussion also: //github.com/scaffold-eth/scaffold-eth-2/issues/369

**Comment by carletex** [comment]
image

I don't want this branch to diverge too much from main, so I think it's a good practice to merge from main once in a while.

The merge it's pretty straightforward (merge from the UI) if non-tricky files are changed (any hook, components, etc) but you have to do it in your machine if there are files like package.json, yarn.lock, template files etc modified. In this case, it was easy, just "accepting ours" for those two files.

**Comment by dgrcode** [comment]
just added the PR https://github.com/scaffold-eth/scaffold-eth-2/pull/432 to remove unused types

**Comment by carletex** [comment]
New npm version published with #432 and #430

**Comment by Pabl0cks** [comment]
Was trying current version of CLI to check how the arguments work, in order to explain them in the Readme, and I'm getting some errors on both my Windows desktop and laptop.

```
E:\GitHub>npx create-eth@latest test22

 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+
 | Create Scaffold-ETH 2 app |
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+

? What solidity framework do you want to use? foundry
? Install packages? Yes


  √ 📁 Create project directory E:\GitHub\test22
  × 🚀 Creating a new Scaffold-ETH 2 app in test22
    → Only URLs with a scheme in: file and data are supported by the default ESM loader. On Windows, absolute paths must be valid file:// URLs.
R…
    📦 Installing dependencies with yarn, this could take a while
    📡 Initializing Git repository
ERROR Error occurred Error [ERR_UNSUPPORTED_ESM_URL_SCHEME]: Only URLs with a scheme in: file and data are supported by the default ESM loader. On Windows, absolute paths must be valid file:// URLs. Received protocol 'c:'
    at new NodeError (node:internal/errors:399:5)
    at throwIfUnsupportedURLScheme (node:internal/modules/esm/resolve:1059:11)
    at defaultResolve (node:internal/modules/esm/resolve:1135:3)
    at nextResolve (node:internal/modules/esm/loader:163:28)
    at ESMLoader.resolve (node:internal/modules/esm/loader:838:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
    at ESMLoader.import (node:internal/modules/esm/loader:525:22)
    at importModuleDynamically (node:internal/modules/esm/translators:110:35)
    at importModuleDynamicallyCallback (node:internal/process/esm_loader:35:14)
    at file:///C:/Users/Pablo/AppData/Local/npm-cache/_npx/85f97c58a7d429b9/node_modules/create-eth/dist/cli.js:197:63 {
  code: 'ERR_UNSUPPORTED_ESM_URL_SCHEME',
  context: [Object: null prototype] {}
}
Uh oh! 😕 Sorry about that! Exiting...
```

**Comment by carletex** [comment]
@Pabl0cks The error might be related to the changes we merged this week, where we started using templates.

Are using NodeJS 18, right? 

Could you also try going back to e844021c7754b3c48f236af010a4c85bf8efd78b and test locally? If not, go back to ee958945081ef8429f24dcfe26554be643c60536 and test again. This will help identifying where the bug was introduced.

**Comment by Pabl0cks** [comment]
> Are using NodeJS 18, right? 

Yep, v18.15.0

> Could you also try going back to https://github.com/scaffold-eth/scaffold-eth-2/commit/e844021c7754b3c48f236af010a4c85bf8efd78b and test locally? If not, go back to https://github.com/scaffold-eth/scaffold-eth-2/commit/ee958945081ef8429f24dcfe26554be643c60536 and test again. This will help identifying where the bug was introduced.

Sure! I've went back and the first commit where it started failing is:  2599a95a0428832d9e5154328a317158a3c18666

**Comment by carletex** [comment]
> Sure! I've went back and the first commit where it started failing is: https://github.com/scaffold-eth/scaffold-eth-2/commit/2599a95a0428832d9e5154328a317158a3c18666

That's exactly when we started using templates (using .mjs). After some research:

https://stackoverflow.com/a/70057245

image

Similar errors in popular libs:
https://github.com/vitejs/vite/issues/1712
https://github.com/okonet/lint-staged/issues/1054

**A commit that fixed it on Vite:** (not sure if useful for us)
https://github.com/vitejs/vite/commit/33d3cca78607e14e4d0b13e7012d065a94d184b1

@dgrcode Could you take a look?

**Comment by Pabl0cks** [comment]
I've tested it a bit to try to get some extra info - logs from the problem, I think the problem is in that line of code in `copy-template-files.ts`:

`const template = (await import(templateFileDescriptor.path)).default;`

Added this code before that line to get some logging:

```
      const argsFilesPath = extensions
        .map((extension) => {
          const argsFilePath = path.join(
            extensionDict[extension].path,
            argsPath
          );
          const fileExists = fs.existsSync(argsFilePath);
          if (!fileExists) {
            return [];
          }
          writeTextToFile("argsFilePath: "+argsFilePath+"\n",'./output.txt');
          return argsFilePath;
        })
        .flat();


      writeTextToFile("targetDir: "+targetDir+"\n",'./output.txt');
      writeTextToFile("templateFileDescriptor: "+templateFileDescriptor.relativePath.split(templateTargetName)[0]+"\n",'./output.txt');
      writeTextToFile("templateTargetName: "+templateTargetName+"\n",'./output.txt');
      writeTextToFile("templateFileDescriptor.path: "+templateFileDescriptor.path+"\n",'./output.txt');
      
```

Those are the last values I get on the log:

**argsFilePath**: E:\GitHub\scaffold-eth-2-readme-scaffold-config\templates\extensions\hardhat\packages\nextjs\scaffold.config.ts.args.mjs
**templateFileDescriptor.path**: E:\GitHub\scaffold-eth-2-readme-scaffold-config\templates\base\packages\nextjs\.gitignore.template.mjs
**templateTargetName**: .gitignore
**templateFileDescriptor**: \packages\nextjs\
**templateTargetName**: scaffold.config.ts
**targetDir**: E:\GitHub\scaffold-eth-2-readme-scaffold-config\test16
**templateFileDescriptor.path**: E:\GitHub\scaffold-eth-2-readme-scaffold-config\templates\base\packages\nextjs\scaffold.config.ts.template.mjs
**targetDir**: E:\GitHub\scaffold-eth-2-readme-scaffold-config\test16
**templateFileDescriptor.path**: E:\GitHub\scaffold-eth-2-readme-scaffold-config\templates\extensions\hardhat\packages\hardhat\.gitignore.template.mjs
**templateTargetName**: .gitignore
**templateFileDescriptor**: \packages\hardhat\


Not sure if this helps, let me know if you want to make any other test on Windows.

**Comment by carletex** [comment]
@Pabl0cks Thanks for this!

Could you try this PR and see if it fixes it? https://github.com/scaffold-eth/scaffold-eth-2/pull/443

P..S I did it a bit blindly, haven't setup a Windows VM (yet :))

**Comment by Pabl0cks** [comment]
> P..S I did it a bit blindly, haven't setup a Windows VM (yet :))

This is one of those errors annoying to fix, why couldn't Windows act like the rest of OS 😥

Sent the console error logs I'm getting now in #443

**Comment by carletex** [comment]
Published 0.0.11 to npm. @Pabl0cks could you test if everything works as expected on windows?

Also added the `yarn test` alias from https://github.com/scaffold-eth/scaffold-eth-2/pull/445 to both hardhat and foundry.

**Comment by Pabl0cks** [comment]
> @Pabl0cks could you test if everything works as expected on windows?

It's working great on windows! gj all 🚀

**Comment by carletex** [comment]
Closing this in favor of https://github.com/scaffold-eth/create-eth

As we discussed, the CLI version will live in its own repo (GitHub actions already publishing to NPM in that repo)

**Pull request closed without merging**